app/pages/project/workflow-selection.jsx

```diff
@@ -31,7 +29,9 @@ class WorkflowSelection extends React.Component {
     const currentUser = user && user.login;
     const userSelectedWorkflow = (preferences && preferences.preferences) ? this.sanitiseID(preferences.preferences.selected_workflow) : undefined;
     if (userSelectedWorkflow && workflow) {
-      if (!this.state.loadingSelectedWorkflow && userSelectedWorkflow !== workflow.id) {
+      const prevWorkflowID = prevProps.workflow && prevProps.workflow.id;
+      const workflowChanged = workflow.id !== prevWorkflowID;
+      if (workflowChanged && userSelectedWorkflow !== workflow.id) {
```

Comment:
This change isn't loading a new workflow, from preferences, when a user logins in. Possibly because `userSelectedWorkflow` doesn't change at the same time as the stored workflow. This should maybe listen for a change in user ID, or a change in user preferences ID, rather than a change in workflow ID.

Reply:
Resolved by deleting this block of code and checking for a change of preferences instead.